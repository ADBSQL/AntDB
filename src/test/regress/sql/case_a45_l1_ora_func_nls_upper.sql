set grammar to oracle;
select nls_upper('ab cDe') from dual;
select nls_upper('\?!@#$%^&*()sa') from dual;
select nls_upper('aa\?!@#$%^&*()aa') from dual;
select nls_upper('\?!@#ss$%^&*()sC') from dual;
select nls_upper('Test你tEst好teST') as str from dual;
select nls_upper('1234sou') from dual;
select nls_upper(to_char('suv')) from dual;
select nls_upper(to_number('NaN')) from dual;

create table t4test (id int,txt varchar);
insert into t4test values(1,'ASD dasDSdas SADASD aSDSFasDFas');
insert into t4test values(2,nls_upper('aSDa SD a SSD'));

select * from t4test order by id;

update t4test set txt=nls_upper(txt);

select * from t4test order by id;

drop table t4test;
