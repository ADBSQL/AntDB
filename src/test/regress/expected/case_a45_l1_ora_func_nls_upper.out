set grammar to oracle;
select nls_upper('ab cDe') from dual;
 NLS_UPPER('AB CDE')  
----------------------
 AB CDE
(1 row)

select nls_upper('\?!@#$%^&*()sa') from dual;
 NLS_UPPER('\?!@#$%^&*()SA')  
------------------------------
 \?!@#$%^&*()SA
(1 row)

select nls_upper('aa\?!@#$%^&*()aa') from dual;
 NLS_UPPER('AA\?!@#$%^&*()AA')  
--------------------------------
 AA\?!@#$%^&*()AA
(1 row)

select nls_upper('\?!@#ss$%^&*()sC') from dual;
 NLS_UPPER('\?!@#SS$%^&*()SC')  
--------------------------------
 \?!@#SS$%^&*()SC
(1 row)

select nls_upper('Test你tEst好teST') as str from dual;
       STR        
------------------
 TEST你TEST好TEST
(1 row)

select nls_upper('1234sou') from dual;
 NLS_UPPER('1234SOU')  
-----------------------
 1234SOU
(1 row)

select nls_upper(to_char('suv')) from dual;
 NLS_UPPER(TO_CHAR('SUV'))  
----------------------------
 SUV
(1 row)

select nls_upper(to_number('NaN')) from dual;
 NLS_UPPER(TO_NUMBER('NAN'))  
------------------------------
 NAN
(1 row)

create table t4test (id int,txt varchar);
insert into t4test values(1,'ASD dasDSdas SADASD aSDSFasDFas');
insert into t4test values(2,nls_upper('aSDa SD a SSD'));
select * from t4test order by id;
 ID |               TXT               
----+---------------------------------
  1 | ASD dasDSdas SADASD aSDSFasDFas
  2 | ASDA SD A SSD
(2 rows)

update t4test set txt=nls_upper(txt);
select * from t4test order by id;
 ID |               TXT               
----+---------------------------------
  1 | ASD DASDSDAS SADASD ASDSFASDFAS
  2 | ASDA SD A SSD
(2 rows)

drop table t4test;
